Tasks/VsTest/inputparser.ts

```diff
@@ -0,0 +1,481 @@
+}
+
+function getTestPlatformSettings(inputDataContract : idc.InputDataContract) : idc.InputDataContract {
+    const vsTestLocationMethod = tl.getInput('vstestLocationMethod');
+    if (vsTestLocationMethod === utils.Constants.vsTestVersionString) {
+        const vsTestVersion = tl.getInput('vsTestVersion');
+        if (utils.Helper.isNullEmptyOrUndefined(vsTestVersion)) {
+            console.log('vsTestVersion is null or empty');
+            throw new Error('vsTestVersion is null or empty');
+        } else if (vsTestVersion.toLowerCase() === 'toolsinstaller') {
+            tl.debug('Trying VsTest installed by tools installer.');
+            ci.publishEvent({ subFeature: 'ToolsInstallerSelected', isToolsInstallerPackageLocationSet: !utils.Helper.isNullEmptyOrUndefined(tl.getVariable(constants.VsTestToolsInstaller.PathToVsTestToolVariable)) });
+
+            const vsTestPackageLocation = tl.getVariable(constants.VsTestToolsInstaller.PathToVsTestToolVariable);
+            tl.debug('Path to VsTest from tools installer: ' + vsTestPackageLocation);
+
+            // get path to vstest.console.exe
+            const matches = tl.findMatch(vsTestPackageLocation, '**\\vstest.console.exe');
+            if (matches && matches.length !== 0) {
+                inputDataContract.VsTestConsolePath = path.dirname(matches[0]);
+            } else {
+                utils.Helper.publishEventToCi(AreaCodes.TOOLSINSTALLERCACHENOTFOUND, tl.loc('toolsInstallerPathNotSet'), 1041, false);
+                throw new Error(tl.loc('toolsInstallerPathNotSet'));
+            }
+
+            // if Tools installer is not there throw.
+            if (utils.Helper.isNullOrWhitespace(inputDataContract.VsTestConsolePath)) {
+                ci.publishEvent({ subFeature: 'ToolsInstallerInstallationError' });
+                utils.Helper.publishEventToCi(AreaCodes.SPECIFIEDVSVERSIONNOTFOUND, 'Tools installer task did not complete successfully.', 1040, true);
+                throw new Error(tl.loc('ToolsInstallerInstallationError'));
+            }
+
+            ci.publishEvent({ subFeature: 'ToolsInstallerInstallationSuccessful' });
+
+        } else if ((vsTestVersion !== '15.0') && (vsTestVersion !== '14.0')
+            && (vsTestVersion.toLowerCase() !== 'latest')) {
+            throw new Error(tl.loc('vstestVersionInvalid', vsTestVersion));
+        } else if (vsTestLocationMethod === utils.Constants.vsTestVersionString && vsTestVersion === '12.0') {
+            throw (tl.loc('vs2013NotSupportedInDta'));
+        } else {
+            console.log(tl.loc('vsVersionSelected', vsTestVersion));
+            inputDataContract.VsTestConsolePath = getTestPlatformPath(inputDataContract);
+        }
+    } else {
+        // hydra: should it be full path or directory above?
+        inputDataContract.VsTestConsolePath = tl.getInput('vsTestLocation');
+        console.log(tl.loc('vstestLocationSpecified', 'vstest.console.exe', inputDataContract.VsTestConsolePath));
```

Comment:
1. VsTools path -> check what's the path
2. Vstest.console.exe path -> vs 2015 [IDE]
3. vstest.console.exe path -> vs 15.5
4. vstest.console.exe path -> 15.6 > [\Extensions\TestWindow\]